src/hotspot/share/prims/jvmtiTagMap.cpp

```diff
@@ -3298,117 +2972,57 @@ void JvmtiTagMap::follow_references(jint heap_filter,
   VMThread::execute(&op);
 }
 
+// Concurrent GC needs to call this in relocation pause, so after the oops are moved
+// and have their new addresses, the table can be rehashed.
+void JvmtiTagMap::set_needs_processing() {
```

Comment:
>  GC callers shouldn't really have to know what processing we're doing here.

I completely disagree with this. It's extremely important that the GC and Runtime code agrees what this code does and where the GC *must* call it. Knowing the details allows us to skip calling this after mark end, but forces us to call it in relocate starte, when objects should start to move. Though, I don't want to block this review because of this point, so if you still thinks that a non-descriptive name is better then we can argue that separately.

Reply:
Ok, I'll rename it to needs_hashing.